tests/load-tests.php

```diff
@@ -148,6 +150,59 @@ public function test_perflab_render_generator() {
 		$this->assertContains( $expected, $output );
 	}
 
+	public function test_perflab_get_active_and_valid_modules() {
+		global $wp_version;
+
+		// Assert that it returns the empty array for dummy modules.
+		$dummy_active_modules = array( 'javascript/demo-module-1', 'something/demo-module-2', 'images/demo-module-3' );
+		add_filter(
+			'perflab_active_modules',
+			function() use ( $dummy_active_modules ) {
+				return $dummy_active_modules;
+			}
+		);
+		$output = perflab_get_active_and_valid_modules();
+		$this->assertIsArray( $output );
+		$this->assertSame( array(), $output );
+
+		// Assert that it returns the array for modules.
+		if ( $wp_version >= '6.1' ) {
+			$active_modules = require plugin_dir_path( PERFLAB_MAIN_FILE ) . 'default-enabled-modules.php';
+			add_filter(
+				'perflab_active_modules',
+				function() use ( $active_modules ) {
+					return $active_modules;
+				}
+			);
+			$output = perflab_get_active_and_valid_modules();
+
+			// Remove Image module as it will marge in 6.1.
+			$remove_marge_modules = array_shift( $active_modules );
+			$this->assertSame( $active_modules, $output );
+		}
+	}
+
+	public function test_perflab_can_load_module() {
```

Comment:
@mukeshpanchal27 This test and the following `test_if_can_load_file_available_but_module_does_not_marge_in_core` look okay and tests the logic within the `perflab_can_load_module` function, but feels odd that it doesn't call the function directly.

I can see why as the `plugin_dir_path( __FILE__ )` used to require in the files is not filterable in WordPress, so it's currently not possible to use the test modules.

I'm wondering if there is a way we can either introduce a filter for the modules path in the plugin or provide a constant that we can overwrite during tests to make use of the test modules?

I would be interested in getting @felixarntz thoughts on this though to discuss a way forward.

Reply:
@mukeshpanchal27 I agree with @jjgrainger here, I think these tests don't really help since they don't test any part of the production code.

Let's see if we can find a way to test the actual `perflab_can_load_module()` function. Otherwise, we should remove these tests.

Have you tried the approach I suggested in https://github.com/WordPress/performance/pull/390#pullrequestreview-1023696565? I'm not 100% sure this will work, but wondering, can we rely on `..` in those paths to get to the correct test modules?

Reply:
@jjgrainger Thanks for the feedback.

@felixarntz Yes, before I missed something on it and I can't figure it out, but again I reviewed it and it works fine with the `perflab_can_load_module()` function test. The `..` approach is working.

I have update unit test for `perflab_can_load_module()` for review.